src/Utils/force_machine_learning_module.f95

```diff
@@ -685,7 +685,7 @@ subroutine  read_distance_file(distance_matrix, data_dir)
 
        call system_timer('Read Pair Distance and Bulid Covariance')
  !      OPEN(2,status='old',file=trim(data_dir)//'Dist.dat',form='UNFORMATTED')
-       call fread_array_d(1, distance_matrix(1,1), trim(data_dir)//'Dist.dat') 
+       call fread_array_d(1, (/distance_matrix(1,1)/), trim(data_dir)//'Dist.dat') 
```

Comment:
This change I suspect will not work, but I'm not sure if there's a unit test to check it.  As for `fwrite_array_d` above, `fread_array_d` expects a pointer to the beginning of the data. This will (or at least could) create a new temporary array with a copy of the first element of `distance_matrix` which is not the same.  I'm not sure what the proper way to do this is, but I don't think this is it.